src/main/java/com/diffmin/patch/PatchGeneration.java

```diff
@@ -109,6 +108,12 @@ public int getSrcNodeIndex(CtElement srcNode) {
                                         .getFormalCtTypeParameters();
                             case PARAMETER:
                                 return ((CtExecutable<?>) element.getParent()).getParameters();
+                            case CONTAINED_TYPE:
+                                CtCompilationUnit cu =
+                                        element.getFactory()
+                                                .CompilationUnit()
+                                                .getOrCreate((CtType<?>) element);
```

Comment:
I had one suggestion here. Instead of using a factory method to get the compilation unit, should we implement an API for `CtType`s which would allow to fetch the compilation unit directly? The API could be something like this - `CtType.getCompilationUnit()`.